toolchain/semantics/semantics_context.h

```diff
@@ -233,6 +233,9 @@ class SemanticsContext {
   auto CanonicalizeStructType(ParseTree::Node parse_node,
                               SemanticsNodeBlockId refs_id) -> SemanticsTypeId;
 
+  auto CanonicalizeTupleType(ParseTree::Node parse_node,
+                             const llvm::SmallVector<SemanticsTypeId>& type_ids)
+      -> SemanticsTypeId;
```

Comment:
```suggestion
  // Handles canonicalization of tuple types. This may create a new tuple type
  // when it has a new structure, or reference an existing tuple type when it
  // duplicates a prior type.
  auto CanonicalizeTupleType(ParseTree::Node parse_node,
                             const llvm::SmallVector<SemanticsTypeId>& type_ids)
      -> SemanticsTypeId;

```

Should be commented.